x-pack/plugins/reporting/server/core.ts

```diff
@@ -43,65 +44,36 @@ interface ReportingInternalStart {
 export class ReportingCore {
   private pluginSetupDeps?: ReportingInternalSetup;
   private pluginStartDeps?: ReportingInternalStart;
-  private browserDriverFactory?: HeadlessChromiumDriverFactory;
-  private readonly pluginSetup$ = new Rx.ReplaySubject<ReportingInternalSetup>();
-  private readonly pluginStart$ = new Rx.ReplaySubject<ReportingInternalStart>();
+  private readonly pluginSetup$ = new Rx.ReplaySubject<boolean>(); // for pluginHasSetup
+  private readonly pluginStart$ = new Rx.ReplaySubject<ReportingInternalStart>(); // for getPluginStartDeps
   private exportTypesRegistry = getExportTypesRegistry();
+  private config?: ReportingConfig;
 
-  constructor(private config: ReportingConfig) {}
+  constructor() {}
 
   public pluginSetup(reportingSetupDeps: ReportingInternalSetup) {
     this.pluginSetupDeps = reportingSetupDeps;
-    this.pluginSetup$.next(reportingSetupDeps);
+    this.pluginSetup$.next(true);
   }
 
   public pluginStart(reportingStartDeps: ReportingInternalStart) {
     this.pluginStart$.next(reportingStartDeps);
   }
 
-  public pluginHasStarted(): Promise<boolean> {
-    return this.pluginStart$.pipe(first(), mapTo(true)).toPromise();
-  }
-
-  public setBrowserDriverFactory(browserDriverFactory: HeadlessChromiumDriverFactory) {
-    this.browserDriverFactory = browserDriverFactory;
+  public setConfig(config: ReportingConfig) {
+    this.config = config;
+    this.pluginSetup$.next(true);
```

Comment:
Full "plugin setup" now depends on 2 things: the `InternalSetup` deps, and the config.